plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/cache/IcebergCacheKeyProvider.java

```diff
@@ -16,16 +16,20 @@
 import io.trino.filesystem.TrinoInputFile;
 import io.trino.filesystem.cache.CacheKeyProvider;
 
-import java.io.IOException;
 import java.util.Optional;
 
 public class IcebergCacheKeyProvider
         implements CacheKeyProvider
 {
     @Override
     public Optional<String> getCacheKey(TrinoInputFile delegate)
-            throws IOException
     {
-        return Optional.of(delegate.location().path());
+        String path = delegate.location().path();
+        if (path.endsWith(".trinoSchema") || path.contains("/.trinoPermissions/")) {
```

Comment:
Why don't we just disable cache if `FileHiveMetastore` is used?
It seems that a separate flag for caching on coordinator would be useful in general

Reply:
I don't want to add coordinator caching flag unless it's absolutely necessary.
I think any other way around this would be more convoluted, what we have now is simple and easy to understand.